lib/ansible/executor/task_executor.py

```diff
@@ -486,7 +486,10 @@ def _execute(self, variables=None):
             try:
                 result = self._handler.run(task_vars=variables)
             except AnsibleConnectionFailure as e:
-                return dict(unreachable=True, msg=to_text(e))
+                failure_dict = dict(unreachable=True,
+                                    msg=to_text(e),
+                                    _ansible_connection_exception=e)
```

Comment:
do not pass an exception object into the result, this turns into JSON datastructures and we don't want to serialize/deserialize exceptions, specially since you use as a boolean on the other side.

just use existing 'exception' key and assign the error, no need for new code.